src/modules/tensor/cpu/kernel/channel_dropout.cpp

```diff
@@ -0,0 +1,287 @@
+    Rpp32u numThreads = handle.GetNumThreads();
+    uint8_t *scratchBuffer = reinterpret_cast<uint8_t *>(handle.GetInitHandle()->mem.mcpu.scratchBufferHost);
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        T *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        uint8_t *maskPtr = scratchBuffer + batchCount * srcDescPtr->c;
+        int seed = randomSeed ? std::random_device{}() : DROPOUT_FIXED_SEED; // Use a true random seed if requested, otherwise use the fixed seed for deterministic QA
+        std::mt19937 rng(seed + batchCount);
+        std::bernoulli_distribution keepDist(1.0f - dropoutProbability[batchCount]); // Distribution for the probability of keeping (not dropping) a channel
+        bool atLeastOneChannelKept = false;
+        for (Rpp32u c = 0; c < dstDescPtr->c; c++)
+        {
+            maskPtr[c] = keepDist(rng);
+            atLeastOneChannelKept |= maskPtr[c];
+        }
+
+        if (!atLeastOneChannelKept)
+            maskPtr[rng() % dstDescPtr->c] = 1;
+
+        T *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        // Channel Dropout with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            T *srcPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                T *srcPtrTemp, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < bufferLength; vectorLoopCount += 3)
+                {
+                    if constexpr (std::is_same<T, Rpp8s>::value)
+                    {
+                        *dstPtrTempR = maskPtr[0] ? srcPtrTemp[0] : -128;
+                        *dstPtrTempG = maskPtr[1] ? srcPtrTemp[1] : -128;
+                        *dstPtrTempB = maskPtr[2] ? srcPtrTemp[2] : -128;
```

Comment:
@RooseweltMcW  This value of maskPtr[0/1/2] is known outside the vectorLoopCount loop and outside of the roi.xywhROI.roiHeight loop too.
So for the entirety of two loops, *dstPtrTempR could either be srcPtrTemp[0] or -128.
We can decide that in advance and not do the if-conditions inside the rows * columns loops.